openlcs/libs/sc_handler.py

```diff
@@ -0,0 +1,130 @@
+        The value of srpm component from corgi is as following:
+        {
+            'uuid': 'uuid',
+            'type': 'SRPM',
+            'name': 'libcom_err',
+            'version': '1.45.6',
+            'release': '2.el8',
+            'license': '',
+            'arch': '',
+            'synced': False
+        }
+        @params: srpm_dir, destination directory for srpm files' dir
+        """
+        components = []
+        if os.path.isdir(srpm_dir):
+            srpms = glob.glob(srpm_dir + "/*.src.rpm")
+            for srpm in srpms:
+                srpm = srpm.split('/')[-1]
+                rpm_nvr = srpm.replace('.src.rpm', '')
+                nvr = koji.parse_NVR(rpm_nvr)
+                component = self.get_component_flat(nvr, 'SRPM')
+                components.append(component)
+        else:
+            err_msg = f'Not found {srpm_dir}'
+            raise RuntimeError(err_msg) from None
+        return components
+
+    def get_source_of_srpm_component(self, srpm_dir, nvr):
+        """
+        Get the source file or srpm
+        """
+        srpm_file = nvr + '.src.rpm'
+        srpm_filepath = os.path.join(srpm_dir, srpm_file)
+        if os.path.isfile(srpm_filepath):
+            return srpm_filepath
+        else:
+            err_msg = f'Not found {srpm_filepath}'
+            raise RuntimeError(err_msg) from None
+
+    def get_misc_component(self, misc_dir, sc_nvr):
+        """
+        Get the metadata component of the source container.
+        After unpacking the source container, it has some matedata files.
+        The metadata files need to be gathered as one misc component.
+        @params: misc_dir, destination directory for source container metadata
+        @params: sc_nvr, source container nvr
+        """
+        components = []
+        if os.path.isdir(misc_dir):
+            nvr = koji.parse_NVR(sc_nvr)
+            nvr['name'] = nvr.get('name') + '-metadata'
+            component = self.get_component_flat(nvr, 'UNKNOWN')
+            components.append(component)
+        else:
+            err_msg = f'Not found {misc_dir}'
+            raise RuntimeError(err_msg) from None
+        return components
+
+    def get_source_of_misc_component(self, misc_dir, nvr):
+        """
+        Compress the metadata files to tar file
+        """
+        # Named the source tar file that unified with misc component
+        tar_file = nvr + '.tar.gz'
+        # Compress the metadata files to misc nvr tar file
+        try:
+            tar_filepath = os.path.join(os.path.dirname(misc_dir), tar_file)
+            compress_source_to_tarball(tar_filepath, misc_dir)
+        except RuntimeError as err:
+            raise RuntimeError(err) from None
+        return tar_filepath
+
+    def get_container_components(self, srpm_dir, misc_dir, sc_nvr):
+        """
+        Get container components.
+        """
+        srpm_components = self.get_srpm_components(srpm_dir, sc_nvr)
```

Comment:
self. get_srpm_components have try... exception, but here not catch the exception

Reply:
Updated.